rust/lance/src/dataset.rs

```diff
@@ -568,6 +570,85 @@ impl Dataset {
         &self.uri
     }
 
+    async fn modify_existing_manifest_config<F>(&self, version: u64, modifier: F) -> Result<()>
+    where
+        F: FnOnce(&mut Manifest) -> Result<()>,
+    {
+        let manifest_location = self
+            .commit_handler
+            .resolve_version_location(&self.base, version, &self.object_store.inner)
+            .await?;
+
+        let mut manifest = Self::load_manifest(
+            self.object_store.as_ref(),
+            &manifest_location,
+            &self.base,
+            self.session.as_ref(),
+        )
+        .await?;
+
+        modifier(&mut manifest)?;
+
+        modify_existing_manifest_file(
+            self.object_store(),
+            self.commit_handler.as_ref(),
+            &self.base,
+            &mut manifest,
+            None,
+            self.manifest_location.naming_scheme,
+        )
+        .await?;
+
+        Ok(())
+    }
+
+    /// List all commit messages with version numbers
+    pub async fn list_commit_messages(&self) -> Result<Vec<(u64, Option<String>)>> {
+        let versions = self.versions().await?;
+        let mut result = Vec::new();
+        for version in versions {
+            // Convert Version to u64 using the version's value field
+            let version_number = version.version;
+
+            let manifest_location = self
+                .commit_handler
+                .resolve_version_location(&self.base, version_number, &self.object_store.inner)
+                .await?;
+
+            let manifest = Self::load_manifest(
+                self.object_store.as_ref(),
+                &manifest_location,
+                &self.base,
+                self.session.as_ref(),
+            )
+            .await?;
+
+            let commit_message = manifest.config.get(COMMIT_MESSAGE_KEY).cloned();
```

Comment:
The `config` field is meant for values that should be automatically carried forward to new versions, like configuration. I don't think the commit message is appropriate for this field.